kubectl-cloudflow/docker/inspect.go

```diff
@@ -0,0 +1,196 @@
+}
+
+// InspectImage access a remote registry to get the image configuration.
+// If no authentication arguments are passed the credentials from user's previous docker login are used.
+func (opts *InspectOptions) inspectImage(imageNamewithScheme string) (imageData *InspectOutput, retErr error) {
+	ctx, cancel := opts.commandTimeoutContext()
+	defer cancel()
+
+	sys, err := opts.Image.newSystemContext()
+	if err != nil {
+		return nil, err
+	}
+
+	src, err := parseImageSource(ctx, opts.Image, imageNamewithScheme)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing image name %q: %v", imageNamewithScheme, err)
+	}
+
+	defer func() {
+		if err := src.Close(); err != nil {
+			retErr = fmt.Errorf(fmt.Sprintf("(could not close image: %v) ", err))
+		}
+	}()
+
+	rawManifest, _, err := src.GetManifest(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("Error retrieving manifest for image: %v", err)
+	}
+
+	img, err := image.FromUnparsedImage(ctx, sys, image.UnparsedInstance(src, nil))
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing manifest for image: %v", err)
+	}
+
+	imgInspect, err := img.Inspect(ctx)
+	if err != nil {
+		return nil, err
+	}
+	outputData := InspectOutput{
+		Name: "", // Set below if DockerReference() is known
+		Tag:  imgInspect.Tag,
+		// Digest is set below.
+		Created:       imgInspect.Created,
+		DockerVersion: imgInspect.DockerVersion,
+		Labels:        imgInspect.Labels,
+		Architecture:  imgInspect.Architecture,
+		Os:            imgInspect.Os,
+		Layers:        imgInspect.Layers,
+		Env:           imgInspect.Env,
+	}
+	outputData.Digest, err = manifest.Digest(rawManifest)
+	if err != nil {
+		return nil, fmt.Errorf("Error computing manifest digest: %v", err)
+	}
+	if dockerRef := img.Reference().DockerReference(); dockerRef != nil {
+		outputData.Name = dockerRef.Name()
+	}
+
+	return &outputData, nil
+}
+
+// commandTimeoutContext returns a context.Context and a cancellation callback based on opts.
+// The caller should usually "defer cancel()" immediately after calling this.
+func (opts *InspectOptions) commandTimeoutContext() (context.Context, context.CancelFunc) {
+	ctx := context.Background()
+	var cancel context.CancelFunc = func() {}
+	if opts.commandTimeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, opts.commandTimeout)
+	}
+	return ctx, cancel
+}
+
+// newSystemContext returns a *types.SystemContext corresponding to opts.
+// It is guaranteed to return a fresh instance, so it is safe to make additional updates to it.
+func (opts *ImageOptions) newSystemContext() (*types.SystemContext, error) {
+	ctx := &types.SystemContext{
+		DockerCertPath:           opts.dockerCertPath,
+		OCISharedBlobDirPath:     opts.sharedBlobDir,
+		AuthFilePath:             opts.authFilePath,
+		DockerDaemonHost:         opts.dockerDaemonHost,
+		DockerDaemonCertPath:     opts.dockerCertPath,
+	}
+
+	if opts.credsOption.present {
+		var err error
+		ctx.DockerAuthConfig, err = getDockerAuth(opts.credsOption.value)
```

Comment:
How does this work? where do the credsOption come from?

Reply:
This an option in the API that we might want to use in the future not used at the moment. For example if we want to pass directly password and username the api is there. I added it because we had a discussion about username and password now passed directly in deploy for the old image format and was not sure if we want this as well in the future. If we see we dont need it we can remove it.